src/test_l1_dsFPD.c

```diff
@@ -353,14 +352,17 @@ void test_l1_dsFPD_positive_dsSetFPState (void)
     // Step 02: Call dsSetFPState() with valid inputs
     result = dsSetFPState(dsFPD_INDICATOR_POWER, dsFPD_STATE_OFF);
     UT_ASSERT_EQUAL(result, dsERR_NONE);
-    
-    // Step 03: Call dsSetFPState() and loop through all valid indicators from kIndicators
-    for (int i = 0; i < sizeof(kIndicators) / sizeof(kIndicators[0]); ++i)
-    {
-        UT_LOG("\n In %s , Indicator: [%d]\n", __FUNCTION__, kIndicators[i].id);
-        result = dsSetFPState(kIndicators[i].id, dsFPD_STATE_ON);
+
+     // Step 03: Call dsSetFPState() and loop through all valid indicators from yaml
+     count = UT_KVP_PROFILE_GET_UINT32("dsFPD/Number_of_Indicators");
+     for (int i = 0; i < count; ++i)
+     {
+        snprintf(buffer, DS_FPD_KEY_SIZE, "dsFPD/SupportedFPDIndicators/%d/Indicator_Type", i);
```

Comment:
No 0 indicator type exists..as it starts from 1.

Reply:
Done